Tiles renames and documentation

groud opened the pull request:
This PR is mainly fills in the documentation for tiles-related stuff. However, while working on it, I spotted a bug and a few things to rename, so I put it into the PR too. 

~~This is a draft for now as I need https://github.com/godotengine/godot/pull/52277 to be merged first.~~

groud committed:
Tiles renames/bugfixing and documentation

mhilbrunner commented on doc/classes/TileData.xml:
```diff
@@ -11,70 +11,81 @@
 			<argument index="0" name="layer_id" type="int" />
 			<argument index="1" name="polygon_index" type="int" />
 			<description>
+				Returns if one-way collisions are enabled for the polygon at index [code]polygon_index[/code] for TileSet physics layer with index [code]layer_id[/code].
```
```suggestion
				Returns whether one-way collisions are enabled for the polygon at index [code]polygon_index[/code] for TileSet physics layer with index [code]layer_id[/code].
```

mhilbrunner commented on doc/classes/TileData.xml:
```diff
@@ -99,59 +112,55 @@
 			<argument index="1" name="polygon_index" type="int" />
 			<argument index="2" name="polygon" type="PackedVector2Array" />
 			<description>
+				Set the points of the polygon at index [code]polygon_index[/code] for TileSet physics layer with index [code]layer_id[/code].
```
```suggestion
				Sets the points of the polygon at index [code]polygon_index[/code] for TileSet physics layer with index [code]layer_id[/code].
```

mhilbrunner commented on doc/classes/TileData.xml:
```diff
@@ -99,59 +112,55 @@
 			<argument index="0" name="layer_id" type="int" />
 			<argument index="1" name="polygons_count" type="int" />
 			<description>
+				Set the polygons count for TileSet physics layer with index [code]layer_id[/code].
```
```suggestion
				Sets the polygons count for TileSet physics layer with index [code]layer_id[/code].
```

mhilbrunner commented on doc/classes/TileData.xml:
```diff
@@ -99,59 +112,55 @@
 			<argument index="0" name="layer_name" type="String" />
 			<argument index="1" name="value" type="Variant" />
 			<description>
+				Set the tile's custom data value for the TileSet custom data layer with name [code]layer_name[/code].
```
```suggestion
				Sets the tile's custom data value for the TileSet custom data layer with name [code]layer_name[/code].
```

mhilbrunner commented on doc/classes/TileData.xml:
```diff
@@ -99,59 +112,55 @@
 			<argument index="0" name="layer_id" type="int" />
 			<argument index="1" name="value" type="Variant" />
 			<description>
+				Set the tile's custom data value for the TileSet custom data layer with index [code]layer_id[/code].
```
```suggestion
				Sets the tile's custom data value for the TileSet custom data layer with index [code]layer_id[/code].
```

mhilbrunner commented on doc/classes/TileData.xml:
```diff
@@ -99,59 +112,55 @@
 			<argument index="0" name="layer_id" type="int" />
 			<argument index="1" name="navigation_polygon" type="NavigationPolygon" />
 			<description>
+				Set the navigation polygon for the TileSet navigation layer with index [code]layer_id[/code].
```
```suggestion
				Sets the navigation polygon for the TileSet navigation layer with index [code]layer_id[/code].
```

mhilbrunner commented on doc/classes/TileData.xml:
```diff
@@ -99,59 +112,55 @@
 			<argument index="0" name="layer_id" type="int" />
 			<argument index="1" name="occluder_polygon" type="OccluderPolygon2D" />
 			<description>
+				Set the occluder for the TileSet occlusion layer with index [code]layer_id[/code].
```
```suggestion
				Sets the occluder for the TileSet occlusion layer with index [code]layer_id[/code].
```

mhilbrunner commented on doc/classes/TileSetScenesCollectionSource.xml:
```diff
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="TileSetScenesCollectionSource" inherits="TileSetSource" version="4.0">
 	<brief_description>
+		Exposes a set of scenes as tiles for a TileSet resource.
```
```suggestion
		Exposes a set of scenes as tiles for a [TileSet] resource.
```

mhilbrunner reviewed (changes requested):
Looks good. A few nitpicks below. :)

groud commented:
Alright, I addressed the feedback and added the missing documentation for TileSetScenesCollectionSource I forgot.
Let me know if anything is still missing :).

mhilbrunner reviewed (approved):
Looks good now. :)

mhilbrunner commented:
Thanks!
